### ISSUE TITLE
Expand dynamic function use in needextend

### ISSUE THREAD

**Issue by chrisjsewell** [useblocks/sphinx-needs#1064]
In #1052 the logic of `needextend` was improved to interpret values beginning/ending with brackets `[[...]]` as dynamic functions

There has been a request to expand this functionality to also allow dynamic functions in a "delimited list", e.g. `ID001;[[..]];ID002`

See comment by @arwedus in https://github.com/useblocks/sphinx-needs/issues/970#issuecomment-1784047828_

**Comment by arwedus** [comment]
good idea to make this a new ticket. I'm in favor of releasing sphinx-needs 1.4 this week over waiting until next week just to add this feature (if that's the choice we'd have to make here).

Apart from the documentation. I think the documentation should always be up to date.

**Comment by chrisjsewell** [comment]
@arwedus, just responding to your comments from the previous issue:

> For me, it is kind of un-intuitive that the filter function is treated differently than a dynamic sphinx-needs function, as it does the same things. Is there a way to treat filter just as other dynamic functions that return needs?

There is no `filter` function, #1052 simply allows the use of dynamic functions as values in the standard manner.

I guess you are suggesting adding an additional "builtin" function to https://github.com/useblocks/sphinx-needs/blob/f1769b63ea02eea0e7933df4c86356bb3ef866e7/sphinx_needs/functions/common.py? 

> And second, I'd assume that it would be possible to mix direct need references with dynamic function calls for need references, but it seems not to be the case. Is this intentional? Could you add this feature, too?

Currently, from #1052, avalue for a link type is only denoted as dynamic if it starts with `[[` and ends with `]]`.
To allow for "delimited lists", we would need to do some form of regex detection/splitting
